src/io/usi.rs

```diff
@@ -122,8 +122,17 @@ impl FromStr for USI {
                             }
                         };
 
-                        this.interpretation = tokens.next().map(|s| s.to_string());
-                        this.provenance = tokens.next_back().map(|s| s.to_string());
+                        let tail = tokens.next().map_or((None, None), |tail| {
+                            const REPOSITORY_CODES: &[&str] = &["PR", "PA", "MA", "JP", "IP", "PP"];
+                            for code in REPOSITORY_CODES {
+                                if let Some((i, p)) = tail.split_once(&format!(":{code}-")) {
```

Comment:
Should this be [`tail.rsplit_once`](https://doc.rust-lang.org/std/string/struct.String.html#method.rsplit_once)?

Reply:
If it is `rsplit_once`, then we can do a simple test if the first chunk ends with an opening brace character, then we know it *must* be part of a ProForma tag and not the start of a provenance ID, correct?

I'll sit with this overnight and review the specification again in the morning.